metricflow/errors/errors.py

```diff
@@ -80,3 +80,12 @@ def __init__(self) -> None:  # noqa: D
 
 class InferenceError(Exception):
     """Exception to represent errors related to inference."""
+
+class BigQueryApproximatePercentileError(RuntimeError):
```

Comment:
I prefer to avoid engine-specific errors. Shall we generalize this to something like `UnsupportedEngineFeatureError` instead? The message would have to go back to being passed in at the callsite, as it was before.

This also allows us to use the same custom exception for all of the various percentile cases we have to deal with. For example, [Databricks](https://github.com/dbt-labs/metricflow/blob/main/metricflow/sql/render/databricks.py#L27-L35), [Snowflake](https://github.com/dbt-labs/metricflow/blob/main/metricflow/sql/render/snowflake.py#L39-L43), [Redshift](https://github.com/dbt-labs/metricflow/blob/main/metricflow/sql/render/redshift.py#L31-L40), and [Postgres](https://github.com/dbt-labs/metricflow/blob/main/metricflow/sql/render/postgres.py#L63-L72) all have different levels of support for different percentile aggregation configurations, so we should probably throw the same error for each one.

Reply:
+1 for `UnsupportedEngineFeatureError`